layers/+lang/c-c++/packages.el

```diff
@@ -254,3 +260,46 @@
     :post-init
     (dolist (mode c-c++-modes)
       (spacemacs/set-leader-keys-for-major-mode mode "gi" 'cscope-index-files))))
+
+;; BEGIN LSP BACKEND PACKAGES
+
+;; See also https://github.com/cquery-project/cquery/wiki/Emacs
+(defun c-c++/init-cquery ()
+  (use-package cquery
+    :if (eq c-c++-backend 'lsp-cquery)
+    :defer t
+    :commands lsp-cquery-enable
+    :init
+    (add-hook 'c-mode-common-hook #'spacemacs//c-c++-lsp-enable)
+    :config
+    (spacemacs//c-c++-lsp-config)))
+
+;; See also https://github.com/MaskRay/ccls/wiki/Emacs
+(defun c-c++/init-ccls ()
+  (use-package ccls
+    :if (eq c-c++-backend 'lsp-ccls)
+    :defer t
+    :commands lsp-ccls-enable
+    :init
+    (add-hook 'c-mode-common-hook #'spacemacs//c-c++-lsp-enable)
+    :config
+    (spacemacs//c-c++-lsp-config)))
+
+;;Intentionally adding both cquery and ccls cache dirs to ignore list, to facilitate switching between
+;;two without multiple caches polluting projectile find file results
+(defun c-c++/pre-init-projectile ()
+  (spacemacs|use-package-add-hook projectile
+    :post-config
+    (progn
+      (add-to-list 'projectile-globally-ignored-directories ".cquery_cached_index")
+      (add-to-list 'projectile-globally-ignored-directories ".ccls-cache")
+      (when c-c++-lsp-cache-dir
+        (add-to-list 'projectile-globally-ignored-directories c-c++-lsp-cache-dir))
+      (when c-c++-adopt-subprojects
+        (setq projectile-project-root-files-top-down-recurring
+          (append '("compile_commands.json"
```

Comment:
For the root indicators, you may want to join discussions here (beyond this PR)

https://github.com/emacs-lsp/lsp-mode/issues/293#issuecomment-423728467